torch/distributed/launch.py

```diff
@@ -227,10 +230,16 @@ def main():
         current_env["LOCAL_RANK"] = str(local_rank)
 
         # spawn the processes
-        cmd = [sys.executable, "-u"]
-
-        if args.module:
-            cmd.append("-m")
+        cmd = []
+        if not args.no_python:
```

Comment:
Can you flip the logic here? So we'd have a `--with-python` flag that defaults to `True`, to avoid the double negation here. It would become:

```
if args.with_python:
  ...
```

Reply:
I can't, actually. With argparse, you can't set up an argument which can be used like `--with_python=False`. Unless you want to do something nonstandard like converting the string value to a boolean. I tried `type=bool`, but none of these strings actually evaluated to False after `parse_args()`: "0", "False", "false", etc.

Of course, we could change the default behavior, and require users to opt-in to the `--with_python` flag. But I assume we don't want to break backwards compatibility.

Options:
* I can rename the option from `no_python` to `without_python` if that's clearer
* I can set a local variable `with_python = not args.no_python` in main() so that the if/else becomes more readable

Reply:
Went with the second option

Reply:
Went with the second option